Add deprecation message for FORWARD_MODEL DESIGN2PARAMS usage

flikka opened the pull request:
**Issue**
Resolves #11935 

**Approach**
Adding FORWARD_MODEL DESIGN2PARAMS to deprecation list

<img width="1442" height="472" alt="image" src="https://github.com/user-attachments/assets/cf0f62e3-6389-4cea-b566-6fce200bcb78" />



- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'just rapid-tests'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Add backport label to latest release (format: 'backport release-branch-name')

<!--
Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).
-->

berland commented on src/ert/config/parsing/config_schema_deprecations.py:
```diff
@@ -204,4 +204,10 @@
         "It has been replaced with the path to the directory containing the "
         "configuration file for the experiment.",
     ),
+    DeprecationInfo(
```
If there are scenarios where DESIGN2PARAMS is really needed in a user config, can the user in some way mute the warning?

flikka replied:
> Not sure if we want to remove completely `DESIGN2PARAMS`. Maybe @oyvindeide can comment on this? When it comes to the message itself I thought more in lines:
> 
> ```
> "FORWARD_MODEL DESIGN2PARAMS is still supported, but we recommend using DESIGN_MATRIX instead. 
> Please consider updating your configuration; see the ERT manual for details."
> ```

Ah, I assumed we made DESIGN_MATRIX to replace DESIGN2PARAMS. Could we check with the user-representatives etc? As @berland  points out, if these both are equally supported I guess we need a different way of informing the user. And to be honest, if it is a fully worthy feature, I think it is more appropriate to put information in the manual or similar? Speaking of which, seems like we don't write anything about DESIGN2PARAMS in the manual? https://ert.readthedocs.io/en/latest/search.html?q=DESIGN2PARAMS&check_keywords=yes&area=default

berland replied:
A rogue user could have some rogue code generating the xls file as part of the forward model, and then run DESIGN2PARAMS on it, but it would surprise me if we actually have anyone doing that.

xjules replied:
@flikka additionally would it be possible to print a line that uses `DESIGN_MATRIX` representing equivalent to `DESIGN2PARAMS` usage (ie. with DESIGN_SHEET, DEFAULT_SHEET and the filename)?

xjules replied:
> A rogue user could have some rogue code generating the xls file as part of the forward model, and then run DESIGN2PARAMS on it, but it would surprise me if we actually have anyone doing that.

I think there are "super" users actually doing what you said, but I don't see a way to distinguish between those cases.

flikka commented:
So maybe like this?
<img width="419" height="378" alt="image" src="https://github.com/user-attachments/assets/79b4b75c-2778-4c6d-a630-50de7863262f" />

flikka commented:
> > So maybe like this?
> 
> Looks very good! I have two small suggestions:
> 
>     * Adding this sentence somewhere:
>       Please see https://ert.readthedocs.io/en/latest/reference/configuration/keywords.html#design-matrix for more details on DESIGN_MATRIX
> 
>     * "have been replaced" -> "will be replaced"

Adding a link (with link) in the suggestor requires some changes to the Suggestor and Depcrecations code, is it important enough? I have changed it to "will be replaced", and removed the "example", I think it is good enough to point it out, and refer to the manual. Gets to crowded and messy if we write "the manual" in these boxes, and one more place to maintain it. This is my suggestion:

<img width="419" height="378" alt="image" src="https://github.com/user-attachments/assets/8eb5f3d0-403f-4db1-942a-02e64b6b5f58" />

xjules commented:
> > > So maybe like this?
> > 
> > 
> > Looks very good! I have two small suggestions:
> > ```
> > * Adding this sentence somewhere:
> >   Please see https://ert.readthedocs.io/en/latest/reference/configuration/keywords.html#design-matrix for more details on DESIGN_MATRIX
> > 
> > * "have been replaced" -> "will be replaced"
> > ```
> 
> Adding a link (with link) in the suggestor requires some changes to the Suggestor and Depcrecations code, is it important enough? I have changed it to "will be replaced", and removed the "example", I think it is good enough to point it out, and refer to the manual. Gets to crowded and messy if we write "the manual" in these boxes, and one more place to maintain it. This is my suggestion:
> 
> <img alt="image" width="419" height="378" src="https://private-user-images.githubusercontent.com/3764922/498731110-8eb5f3d0-403f-4db1-942a-02e64b6b5f58.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk5MTAxMTIsIm5iZiI6MTc1OTkwOTgxMiwicGF0aCI6Ii8zNzY0OTIyLzQ5ODczMTExMC04ZWI1ZjNkMC00MDNmLTRkYjEtOTQyYS0wMmU2NGI2YjVmNTgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMDhUMDc1MDEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTdmYWI4OTAxZjEyZDgwMGZmNWQxYTBlZGY4MmQ2YWFlZjNlNmNjODFmZDdkMmJlMTJlZmVmODcwNzIwMmU2ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.qVmyNXs-yq6S3g_NjwKZJQnHPqGbxZEsEUXzj8ZdhRo">

 - There is a type "will been" ...
 - Adding link to docs was suggested by @eivindjahren . We do use it https://github.com/equinor/ert/blob/c8742981dc59773893a361dd0a8d82a89c37a82f/src/ert/config/parsing/config_schema_deprecations.py#L79
